core/src/main/java/com/google/cloud/sql/core/ConnectionMetadata.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+/**
+ * A value object containing configuration needed to set up an mTLS connection to a Cloud SQL
+ * instance.
+ */
+public class ConnectionMetadata {
```

Comment:
We already have a `Metadata` class used by `ConnectionInfo`. Is it possible to merge them and have only one? 

So `ConnectionInfo` would have `ConnectionMetadata` instead of having both `Metadata` and `SslData`.

The use would be: 

```
ConnectionMetadata getConnectionMetadata(long timeoutMs) {
    return getConnectionInfo(timeoutMs).getConnectionMetadata();
}
```

Reply:
I'm open to ideas. This is pretty messy. 

The best solution I could think of is to make the value object `ConnectionMetadata` that extracts only the essential info out these 3 classes plus ConnectionConfig hide this mess from the R2DBC driver. 

Unfortunately, I don't think we can easily combine `ConnectionMetadata`, `Metadata`, `SslData` and `ConnectionInfo` because they are created at different times and build on each other. 

(See DefaultConnectionInfoRepository)

Metadata, SslData, and ConnectionInfo are in separate classes because they get built at different phases of the refresh process. 

First two requests API requests are made simultaneously: 
- Metadata created from the GET InstanceMetadata API request. 
- an ephemereral certificate is created for the instance

Then when both are completed, an SslData is created from the combination of Metadata and ephemeral certificate. 

Finally, a ConnectionInfo is created from the Metadata and SslData.

Reply:
Got it. 

Would it be possible to rename `Metadata` to `InstanceMetadata` ? 

It would be clear that `ConnectionMetadata` is related to the connection and `InstanceMetadata` to the Cloud SQL instance.

It's just a suggestion.

Reply:
Yeah, `InstanceMetadata` seems more clear.